app/user/user_exceptions.c

```diff
@@ -57,50 +55,37 @@ void load_non_32_wide_handler (struct exception_frame *ef, uint32_t cause)
     "ssa8l %2;"              /* set up shift register for src op */
     "src   %1, a6, a5;"      /* right shift to get faulting instruction */
     :"=r"(excvaddr), "=r"(insn)
-    :"r"(epc1)
+    :"r"(ef->epc)
     :"a4", "a5", "a6"
   );
-
-  uint32_t valmask = 0;
-  uint32_t what = insn & LOAD_MASK;
-
-  if (what == L8UI_MATCH)
-    valmask = 0xffu;
-  else if (what == L16UI_MATCH || what == L16SI_MATCH)
-    valmask = 0xffffu;
-  else
-  {
-die:
-    /* Turns out we couldn't fix this, so try and chain to the handler
-     * that was set. (This is typically a remote GDB break). If none 
-     * then trigger a system break instead and hang if the break doesn't 
-     * get handled. This is effectively what would happen if the default 
-     * handler was installed. */
-    if (load_store_handler) {
-      load_store_handler(ef, cause);
-      return;
+ 
+  if ((insn & 0x0000f0u) != 0x10u) {
```

Comment:
I went to the explicit forms because I am doing intimate bit diddling here to shave instructions off hot code. For example when we are checking for 
```
... .... 0000 .... ... 0010b  L8UI
... .... 0001 .... ... 0010b  L16UI
... .... 1001 .... ... 0010b  L16SI
```
where "." denotes don't care checking for L16UI _or_ L16SI is the same as checking `... .... .001 .... ... 0010b` which saves a branch test and a couple of instructions, but since this is the L16 branches which are almost never executed, I am happy to go back to the old symbolic approach and take this hit as you prefer.

Reply:
I didn't mean to imply you had to change the logic, only to suggest human readable names for the constants. If you're really against that, please remove the now unused macro defs up top.